Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -82,31 +103,54 @@ sub initPlugin {
 
 sub shutdownPlugin {
 	my @played = values %recentlyPlayed;
+
 	$prefs->set('recent', \@played);
 }
 
+sub updateRecentlyPlayed {
+	my ($class, $client, $song) = @_;
+	my ($url) = unwrap($song->originUrl);
+
+	$recentlyPlayed{$url} ||= { 
+			url      => $url,
+			title    => $song->track->title,
+			cover    => Slim::Player::ProtocolHandlers->iconForURL($url, $client),
+			duration => $song->duration,
+	};	
+}
+
+sub unwrap {
+	return shift =~ m|^podcast://([^&]+)(?:&from=(\d+)$)?|;	
+}
+
+sub wrap {
+	my ($url, $from) = @_;
+	
+	return 'podcast://' . $url . ($from ? "&from=$from" : '');
```

Comment:
URI escape `$url` to protect a potential `&from...` in there?

Reply:
That has nasty side effects in term of caching and retrieving the url (and track management). I'll have to better think about it

Reply:
As long as it's un/-escaped consistently there wouldn't be any issue, would there?

Reply:
Unfortunately not fully. The regex in info.pm expects an unescaped uri (like https?://), so I need the OPML list to be unescaped otherwise the "remote_image_" will not be cached. I'll do a different pattern like "[from=xxx]" so that there is no risk of conflict

Reply:
done

Reply:
See my comment in Info.pm. I don't like the change there, as it's rather plugin specific in a globally used function. I'd prefer to do the caching here, in the plugin code. Then there wouldn't be any problem escaping the URL either. Because not escaping it can lead to issues.

Reply:
Why globally used function? You mean wrapUrl or the info.pm change? If this is info.pm, I understand your point but it's not fully one plugin specific change, it's more generic for "thing" plugins. Now candidly, I've tried very hard to avoid that change but because of the issue of following redirects and keeping the link with the initial $url's metadata, I really don't know how to "carry forward" that information - this is because scanUrl does change the $track and its url and changing things there is frightening

Reply:
I've pushed an update that I hope (and think) you'll like better. It simply pushed the OPML item's image to remote_image_$url for every ûrl whose PH declares it's remote. I can see the benefit that when a track is added, the PH can immediately display an icon without having to gather all metadata. It could even do that only when the track is actually playing